Mathlib/Analysis/Asymptotics/TVS.lean

```diff
@@ -105,6 +105,29 @@ theorem isLittleOTVS_map {f : α → E} {g : α → F} {k : β → α} {l : Filt
     f =o[𝕜; map k l] g ↔ (f ∘ k) =o[𝕜;l] (g ∘ k) := by
   simp [IsLittleOTVS]
 
+theorem IsLittleOTVS.add [TopologicalAddGroup E] [ContinuousSMul 𝕜 E]
+    {f₁ f₂ : α → E} {g : α → F} {l : Filter α}
+    (h₁ : f₁ =o[𝕜; l] g) (h₂ : f₂ =o[𝕜; l] g) : (f₁ + f₂) =o[𝕜; l] g := by
+  intro U' hU'
+  obtain ⟨U, hU, hUb, hUU'⟩ : ∃ U ∈ 𝓝 (0 : E), Balanced 𝕜 U ∧ U + U ⊆ U' := by
+    rcases exists_open_nhds_zero_add_subset hU' with ⟨U₁, hU₁o, hU₁, hU₁U'⟩
+    rcases (nhds_basis_balanced 𝕜 E).mem_iff.mp (hU₁o.mem_nhds hU₁) with ⟨U, ⟨hU₀, hUb⟩, hUU₁⟩
```

Comment:
nit: I'd perhaps use `obtain` on all three liens for consistency.

Reply:
Another option is to have a lemma that turns a `(nhds 0).HasBasis p s` into `(nhds 0).HasBasis p (fun i => s i + s i)`, then use `HasBasis.isLittleOTVS_iff` for `(nhds_basis_balanced K E).newLemma`.

Reply:
As far as I can tell your argument here only works in one direction; maybe the other direction is also trivially true using a lemma I haven't found:
```lean
theorem Filter.HasBasis.add_self {ι : Sort*} [TopologicalAddGroup E] [ContinuousSMul 𝕜 E]
    {p : ι → Prop} {s : ι → Set E}
    (h : (𝓝 0).HasBasis p s) :
    (𝓝 0).HasBasis p fun i => s i + s i := by
  constructor
  intro U
  constructor
  · intro hU
    rcases exists_open_nhds_zero_add_subset hU with ⟨U₁, hU₁o, hU₁, hU₁U'⟩
    obtain ⟨i, hi, hs⟩ := h.mem_iff.1 (hU₁o.mem_nhds hU₁)
    refine ⟨i, hi, add_subset_add hs hs |>.trans hU₁U'⟩
  · rintro ⟨i, hi, hs⟩
    rw [h.mem_iff]
    refine ⟨i, hi, subset_add_left _ ?_ |>.trans hs⟩
    sorry
```

Reply:
Nevermind, it's trivial. I'll push what I have and let you golf it.